src/main/java/dk/cngroup/lentils/service/ProgressService.java

```diff
@@ -133,4 +138,40 @@ private int getStageOfPendingCypherForTeam(final Team team) {
                 .get();
         return statusPending.getCypher().getStage();
     }
+
+    public List<TeamProgressWithTeam> getAllTeamsWithTeamProgress() {
+        List<Team> teams = teamService.getAll();
+        List<TeamProgressWithTeam> teamsProgresses = teams.stream()
+                .map(team -> new TeamProgressWithTeam(team, this.getTeamProgress(team)))
+                .collect(Collectors.toList());
+        return teamsProgresses;
+    }
+
+    public List<TeamProgressWithTeam> getSearchedTeamsWithTeamProgress(final String searchString) {
+        List<Team> teams = this.getSearchedTeams(searchString);
+        List<TeamProgressWithTeam> teamsProgresses = teams.stream()
+                .map(team -> new TeamProgressWithTeam(team, this.getTeamProgress(team)))
+                .collect(Collectors.toList());
+        return teamsProgresses;
+    }
+
+    public String getTeamProgress(final Team team) {
```

Comment:
Ta metoda by lepe vypadala nejak takto:
```java
public String getTeamProgress(final Team team) {
	if (statusService.getAllByTeam(team).isEmpty()) {
		return new TeamProgressNotStarted();
	}

	List<Status> pendingCyphers = statusService.getPendingCyphers(team);
	if (pendingCyphers.isEmpty()) {
    		return new TeamProgressFinished();
	}

	int stageOfPendingCypher = pendingCyphers.get(0).getCypher().getStage();
	return new TeamProgressPlaying(stageOfPendingCypher);
}	
```
A bylo by dobre ji mit v `TeamProgresFactory` a mohla by se jmenovat `create()` a byt staticka.

Reply:
máš pravdu, jenom jsem do toho returnu musela všude dát .toString().

Reply:
opraveno

Reply:
tak jsem si nevšimla toho posledního komentáře - tak to jdu ještě udělat...

Reply:
Nemusela. Ono by bylo lepsi kdyby metoda `getTeamProgress` vracela opravd objekt TeamProgress a ne string. Uz podle nazvu. A tim padem by nebylo spatne kdyzby pak i TeamProgressWithTeam byly entity Team a TeamProgress. (A ne Team a String.)

Reply:
tak upraveno snad už naposledy